File: files/2026-02-15/CSEC 15022026.R
library(tidyverse)
library(readxl)

path <- "2026-02-15/Challenge 103.xlsx"
input <- read_excel(path, range = "B2:B13")
test <- read_excel(path, range = "D2:D13")

result <- input %>%
  mutate(
    n = row_number() - 1,
    .by = Invoice,
    `Suffixed Invoice` = if_else(
      n == 0,
      Invoice,
      paste0(Invoice, "_D", str_extract(Invoice, "\\d+"), "-", n)
    )
  ) %>%
  select(`Suffixed Invoice`)

all.equal(result$`Suffixed Invoice`, test$`Suffixed Invoice`)
# Inconsitent second delim in test data, but otherwise correct.

File: files/2026-02-15/CSEC 15022026.py
import pandas as pd

path = "2026-02-15/Challenge 103.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=12)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=12)

c = input.groupby("Invoice").cumcount()
result = pd.DataFrame({
    "Suffixed Invoice": input["Invoice"].where(
        c.eq(0),
        input["Invoice"] + "_D" +
        input["Invoice"].str.extract(r"(\d+)")[0] +
        "-" + c.astype(str)
    )
})

print(result.equals(test))
# Inconsitent second delim in test data, but otherwise correct.
